cloudinit/config/cc_refresh_rmc_and_interface.py

```diff
@@ -0,0 +1,114 @@
+# (c) Copyright IBM Corp. 2020 All Rights Reserved
+#
+# Author: Aman Kumar Sinha <amansi26@in.ibm.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+"""
+Refresh IPv6 interface and RMC
+------------------------------
+**Summary:** Ensure Network Manager is not managing IPv6 interface
+
+This module is IBM PowerVM Hypervisor specific
+
+Reliable Scalable Cluster Technology (RSCT) is a set of software components
+that together provide a comprehensive clustering environment(RAS features)
+for IBM PowerVM based virtual machines. RSCT includes the Resource
+Monitoring and Control (RMC) subsystem. RMC is a generalized framework used
+for managing, monitoring, and manipulating resources. RMC runs as a daemon
+process on individual machines and needs creation of unique node id and
+restarts during VM boot.
+More details refer
+https://www.ibm.com/support/knowledgecenter/en/SGVKBA_3.2/admin/bl503_ovrv.htm
+
+This module handles
+- Refreshing RMC
+- Disabling NetworkManager from handling IPv6 interface, as IPv6 interface
+  is used for communication between RMC daemon and PowerVM hypervisor.
+
+**Internal name:** per always
+
+**Supported distros:** RHEL
+
+"""
+
+from cloudinit import log as logging
+from cloudinit.settings import PER_ALWAYS
+from cloudinit import util
+from cloudinit import netinfo
+from cloudinit import subp
+
+frequency = PER_ALWAYS
+
+LOG = logging.getLogger(__name__)
+# Ensure that /opt/rsct/bin has been added to standard PATH of the
+# distro. The symlink to rmcctrl is /usr/sbin/rsct/bin/rmcctrl .
+RMCCTRL = 'rmcctrl'
+SED = 'sed'
+
+
+def handle(name, _cfg, _cloud, log, _args):
+    info = netinfo.netdev_info()
+    # Setting NM_CONTROLLED=no for IPv6 interface
+    # making it down and up
+    info_len = len(info.keys())
+
+    LOG.debug('Making the IPv6 up explicitly. '
+              'Ensuring IPv6 interface is not being handled by NetworkManager'
+              ' and it is  restarted to re-establish the communication with '
+              'the hypervisor')
+    for info_count in range(info_len):
+        try:
```

Comment:
since you are calling list(info.keys()) so many times, just call it once and store the result.

    info_keys = list(info.keys())

I think this loop can be cleaned up a bunch.


```python

for iface, data in info.items():
    if iface == "lo":
        # i think this is what you were after
        continue
    if len(data.get("ipv4", []) != 0:
        # skip this interface, as it has ipv4 addrs
         continue
    disable_NM_ipv6(iface)
```